src/utils.jl

```diff
@@ -107,3 +118,65 @@ end
   isnothing(ret) && return IteratorReturn(nothing)
   ret
 end
+
+# adapted from Base to handle world age
+function code_typed_by_type(@nospecialize(tt::Type);
+                            optimize::Bool=true,
+                            world::UInt=Base.get_world_counter(),
+                            interp::Core.Compiler.AbstractInterpreter=Core.Compiler.NativeInterpreter(world))
+    tt = Base.to_tuple_type(tt)
+    match, valid_worlds = Core.Compiler.findsup(tt, Core.Compiler.InternalMethodTable(world))
+    meth = Base.func_for_method_checked(match.method, tt, match.sparams)
+    frame = Core.Compiler.typeinf_frame(interp, meth, match.spec_types, match.sparams, optimize)
```

Comment:
@aviatesk Should we do anything with the valid_worlds of the frame? I think @topolarity suggested we should probably intersect the frame worlds with the lookup worlds, but we weren't totally sure.

Reply:
Good point. We probably want to add
```suggestion
    frame = Core.Compiler.typeinf_frame(interp, meth, match.spec_types, match.sparams, optimize)
    valid_worlds = Core.Compiler.intersect(valid_worlds, frame.valid_worlds)
```